### PR TITLE
feat: middleware datacite

### PR DESCRIPTION
**Pull request by felixleo22**

### CODE REVIEW

File: datacite/index.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+   * @param {Object} result the document used to enrich the EC
+   */
+  function enrichEc(ec, result) {
+    const titleData = result.attributes.titles.find((res) => {
+      return res && res.title;
+    });
+
+    if (titleData) {
+      ec['publication_title'] = titleData.title;
+    }
```
Comment (nojhamster) [inline datacite/index.js:142]: An exception may be thrown if the response is not as expected.

```suggestion
    const titles = result && result.attributes && result.attributes.titles;
    if (!Array.isArray(titles)) { return; }
    const titleData = titles.find((item) => (item && item.title));

    if (titleData) {
      ec['publication_title'] = titleData.title;
    }
```

File: datacite/manifest.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "headers": [
+    { "name": "datacite-cache" },
+    { "name": "datacite-ttl" },
+    { "name": "datacite-throttle" },
+    { "name": "datacite-paquet-size" },
+    { "name": "datacite-buffer-size" },
+    { "name": "datacite-api-key" }
```
Comment (nojhamster) [inline datacite/manifest.json:8]:
```suggestion
    { "name": "datacite-buffer-size" }
```

File: datacite/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+| journal_is_oa | Boolean | Is this resource published in a completely OA journal. |
+| oa_status | String | The OA status, or color, of this resource. |
+| updated | String | Time when the data for this resource was last updated. |
+### Example :
+
+```bash
+curl -v -X POST http://localhost:59599
+  -H "ezPAARSE-Middlewares: ezunpaywall"
+  -F "files[]=@access.log"
+```
```
Comment (nojhamster) [inline datacite/README.md:28]: This Readme should be updated for Datacite instead of ezUnpaywall

**Review by nojhamster — CHANGES_REQUESTED** [review]

**Comment by felixleo22** [comment]
It's done

**Pull request merged**